app/Http/Controllers/Bible/VideoStreamController.php

```diff
@@ -70,48 +88,40 @@ function () use ($arclight_id) {
-            if ($verse && isset($verse[$book->id_osis][$chapter])) {
-                $streaming_component_responses[$verse_key] = $arclight_service->doRequest(
-                    'media-components/' . $verse_key . '/languages/' . $arclight_id,
-                    $arclight_id,
-                    false
-                );
-            }
-        }
-
         try {
+            $media_components_response = $arclight_service->doRequest(
+                'media-components',
+                $arclight_id,
+                false,
+                'metadataLanguageTags=' . $metadata_language_tag . ',en&ids='.join(',', $verse_keys)
+            );
+
             $media_components = $arclight_service->getContent($media_components_response);
-        } catch (Exception $e) {
-            return strpos($e, 'timed out') !== false
-                ? $this->setStatusCode(408)->replyWithError('Request timeout')
-                : $this->setStatusCode(500)->replyWithError('Internal server error');
-        }
 
-        $films = $this->getJesusFilmsFromMediaComponents($media_components, $verses, $book->id_osis, $chapter);
+            $films = $this->getJesusFilmsFromMediaComponents($media_components, $verses, $book->id_osis, $chapter);
 
-        foreach ($streaming_component_responses as $verse_key => $response) {
-            if (isset($films[$verse_key])) {
-                try {
-                    $streaming_component = $arclight_service->getContent($response);
-                } catch (Exception $e) {
-                    return strpos($e, 'timed out') !== false
-                        ? $this->setStatusCode(408)->replyWithError('Request timeout')
-                        : $this->setStatusCode(500)->replyWithError('Internal server error');
+            // We don't cache this portion because streaming url session can expire
+            foreach ($verses as $verse_key => $verse) {
+                if ($verse && isset($verse[$book->id_osis][$chapter])) {
+                    $streaming_component_responses[$verse_key] = $arclight_service->doRequest(
+                        'media-components/' . $verse_key . '/languages/' . $arclight_id,
+                        $arclight_id,
+                        false
+                    );
                 }
+            }
 
-                $films[$verse_key]['meta']['file_name'] = $streaming_component->streamingUrls->m3u8[0]->url;
+            foreach ($streaming_component_responses as $verse_key => $response) {
+                if (isset($films[$verse_key])) {
+                    $streaming_component = $arclight_service->getContent($response);
+                    $films[$verse_key]['meta']['file_name'] = $streaming_component->streamingUrls->m3u8[0]->url;
+                }
             }
+        } catch (Exception $e) {
+            \Log::channel('errorlog')
+                ->error(["Arclight Service - Error: '{$e->getMessage()}" ]);
+            return [];
```

Comment:
@bradflood For now on, It will avoid return a 500 or 401 status code when arclight API returns a status code different to 200 status code.